client/cmd/job/change_namespace.go

```diff
@@ -118,25 +120,272 @@ func (c *changeNamespaceCommand) sendChangeNamespaceRequest(jobName string) erro
 	return err
 }
 
+func getJobParentsFilePathList(fileFS afero.Fs, jobDirPath string) ([]string, error) {
+	var parentsFilePathList []string
+	var errorMsg string
+	splitDirPaths := strings.Split(jobDirPath, "/")
+	for i := range splitDirPaths {
+		pathNearSpecIdx := len(splitDirPaths) - i
+		rootToNearSpecPaths := splitDirPaths[:pathNearSpecIdx]
+		parentDirPath := filepath.Join(rootToNearSpecPaths...)
+		parenFilePath := filepath.Join(parentDirPath, "this.yaml")
+		exist, err := afero.Exists(fileFS, parenFilePath)
+		if err != nil {
+			errorMsg += fmt.Sprintf("Unable to read directory %s to check 'this.yaml', err: %s\n", parenFilePath, err.Error())
+			continue
+		}
+		if exist {
+			parentsFilePathList = append(parentsFilePathList, parenFilePath)
+		}
+	}
+	var err error
+	if len(errorMsg) > 0 {
+		err = errors.New(errorMsg)
+	}
+	return parentsFilePathList, err
+}
+
+func mergeYAMLFiles(fs afero.Fs, sourcePath, targetPath string) error {
```

Comment:
can we merge yaml file without convert it to the object? or we could use 3rd party lib as it's common yaml operation I believe

Reply:
interesting implementation for merging operation from helm repo, https://github.com/helm/helm/blob/e9abdc5efe11cdc23576c20c97011d452201cd92/pkg/cli/values/options.go#L88
can take a look on this for additional reference

Reply:
the codes look similar, also the helm function here only works to merge maps. we will need to test this again if we use helms implementation. But if we could find a library to do this, then that would be great indeed

Reply:
>can we merge yaml file without convert it to the object 

Actually we need to compare the yaml `semantically`, a file comparison would do a line by line comparison

Reply:
got it, I couldn't find the stable 3rd party either, the closest way is utilize the viper, for this case it's not needed. left some suggestion on `mergeValue` function